test/spec/modules/cwireBidAdapter_spec.js

```diff
@@ -138,12 +140,145 @@ describe('C-WIRE bid adapter', () => {
+          cwcreative: 1234,
+          cwapikey: 'api_key_5',
+          refgroups: 'group_5',
+        }).build();
+
+      const bidderRequest01 = new BidderRequestBuilder().build();
+
+      const requests = spec.buildRequests([bid01, bid02], bidderRequest01);
+
+      expect(requests.data.slots.length).to.equal(2);
+      expect(requests.data.cwid).to.be.null;
+      expect(requests.data.cwid).to.be.null;
+      expect(requests.data.cwcreative).to.equal(33);
+      expect(requests.data.cwapikey).to.equal('api_key_33');
+      expect(requests.data.refgroups[0]).to.equal('group_33');
+    });
+
+    it('creates a valid request - read debug params from 3 different slots', function () {
+      const bid01 = new BidRequestBuilder()
+        .withParams({
+          cwcreative: 33,
+        }).build();
+
+      const bid02 = new BidRequestBuilder()
+        .withParams({
+          cwapikey: 'api_key_5',
+        }).build();
+
+      const bid03 = new BidRequestBuilder()
+        .withParams({
+          refgroups: 'group_5',
+        }).build();
+      const bidderRequest01 = new BidderRequestBuilder().build();
+
+      const requests = spec.buildRequests([bid01, bid02, bid03], bidderRequest01);
+
+      expect(requests.data.slots.length).to.equal(3);
+      expect(requests.data.cwid).to.be.null;
+      expect(requests.data.cwid).to.be.null;
+      expect(requests.data.cwcreative).to.equal(33);
+      expect(requests.data.cwapikey).to.equal('api_key_5');
+      expect(requests.data.refgroups[0]).to.equal('group_5');
+    });
+
+    it('creates a valid request - config is overriden by URL params', function () {
       // for whatever reason stub for getWindowLocation does not work
       // so this was the closest way to test for get params
       const params = sandbox.stub(utils, 'getParameterByName');
-      params.withArgs('cwgroups').returns('group_1');
-      params.withArgs('cwcreative').returns('54321');
+      params.withArgs('cwgroups').returns('group_2');
+      params.withArgs('cwcreative').returns('654321');
 
+      const bid01 = new BidRequestBuilder({
+        mediaTypes: {
+          banner: {
+            sizes: [[1, 1]],
+          }
+        }
+      }).withParams({
+        cwcreative: 54321,
+        cwapikey: 'xxx-xxx-yyy-zzz',
+        refgroups: 'group_1',
+      }).build();
+
+      const bidderRequest01 = new BidderRequestBuilder().build();
+
+      const requests = spec.buildRequests([bid01], bidderRequest01);
+
+      expect(requests.data.slots.length).to.equal(1);
+      expect(requests.data.cwid).to.be.null;
+      expect(requests.data.cwid).to.be.null;
+      expect(requests.data.slots[0].sizes[0]).to.equal('1x1');
+      expect(requests.data.cwcreative).to.equal(654321);
+      expect(requests.data.cwapikey).to.equal('xxx-xxx-yyy-zzz');
+      expect(requests.data.refgroups[0]).to.equal('group_2');
+    });
+
+    it('creates a valid request - if config not set null or empty array is sent', function () {
```

Comment:
I don't see an empty array - is the test case name accurate?

Reply:
Ah, bad wording

it was `if config not set null or empty array is sent`

renamed to:
`if params are not set, null or empty array are sent to the API`

and then, there is check for that, nothing changed there:

```javascript
      expect(requests.data.cwcreative).to.equal(null);
      expect(requests.data.cwapikey).to.equal(null);
      expect(requests.data.refgroups.length).to.equal(0);
```

so, if params are not set anywhere, I send null  or empty array